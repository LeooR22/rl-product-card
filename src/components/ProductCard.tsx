import React from 'react';

import { useProduct } from '../hooks/useProduct';

import styles from '../styles/styles.module.css';

import { createContext, CSSProperties /*ReactElement*/ } from 'react';
import { InitialValues, ProductCardHandlers } from '../interfaces/interfaces';
import {
  Product,
  ProductContextProps,
  onChangeArgs,
} from '../interfaces/interfaces';

export const ProductContext = createContext({} as ProductContextProps);
const { Provider } = ProductContext;

export interface Props {
  // children?: ReactElement | ReactElement[];
  children: (args: ProductCardHandlers) => JSX.Element;
  className?: string;
  initialValues?: InitialValues;
  product: Product;
  style?: CSSProperties;
  value?: number;
  onChange?: (args: onChangeArgs) => void;
}

export const ProductCard = ({
  children,
  className,
  product,
  style,
  value,
  initialValues,
  onChange,
}: Props) => {
  const {
    counter,
    increaseBy,
    maxCount,
    isMaxCountReached,
    reset,
  } = useProduct({
    onChange,
    product,
    value,
    initialValues,
  });

  return (
    <Provider
      value={{
        counter,
        increaseBy,
        product,
        maxCount,
      }}
    >
      <div className={`${styles.productCard} ${className}`} style={style}>
        {children({
          count: counter,
          isMaxCountReached,
          maxCount: initialValues?.maxCount,
          product,
          increaseBy,
          reset,
        })}
      </div>
    </Provider>
  );
};
