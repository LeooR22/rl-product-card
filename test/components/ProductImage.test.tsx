import React from 'react';

import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/products';

describe('pruebas en <ProductImage/>', () => {
  test('debe de renderizar correctamente con imagen', () => {
    const wrapper = renderer.create(<ProductImage img={'https://hola.jpg'} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar el componente con la imagen personalizado ', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
