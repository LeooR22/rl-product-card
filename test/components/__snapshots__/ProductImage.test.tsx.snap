// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <ProductImage/> debe de mostrar el componente con la imagen personalizado  1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product "
    className="productImg undefined"
    src="./coffee-mug2.png  "
  />
</div>
`;

exports[`pruebas en <ProductImage/> debe de renderizar correctamente con imagen 1`] = `
<img
  alt="Product "
  className="productImg undefined"
  src="https://hola.jpg"
/>
`;
